python/lsst/ts/FiberSpectrograph/fiberSpectrograph.py

```diff
@@ -29,26 +29,55 @@
 import struct
 
 
-def assert_avs_code(code, msg):
+class AvsReturnError(Exception):
+    """Exception raised if an ``AVS_*`` C function returns an error code.
+
+    Parameters
+    ----------
+    code : `int`
+        The error code returned by the failing function.
+    what : `str`
+        The function that was called that returned this error code.
+    """
+    def __init__(self, code, what):
+        try:
+            self.code = AvsReturnCode(code)
+            self.valid = True
```

Comment:
Please either document the `valid` attribute or make it private and provide a comment. (I can see what it means by reading the code, but still...)